src/main/java/org/terasology/lost/generator/LostWorldGenerator.java

```diff
@@ -90,47 +89,70 @@ protected WorldBuilder createWorld() {
                 .addRasterizer(new WhittakerRasterizer())
                 .addRasterizer(new RiverRasterizer())
                 .addRasterizer(new TreeRasterizer())
-                .addRasterizer(new FloraRasterizer());
-//                .addPlugins();
+                .addRasterizer(new FloraRasterizer())
+                .addPlugins();
     }
 
     @Override
     public Vector3f getSpawnPosition(EntityRef entity) {
         LocationComponent loc = entity.getComponent(LocationComponent.class);
         Vector3f pos = loc.getWorldPosition();
-
-        int searchRadius = 500;
-        Vector3i ext = new Vector3i(searchRadius, 1, searchRadius);
+        Vector3i ext = new Vector3i(SEARCH_RADIUS, 1, SEARCH_RADIUS);
         Vector3i desiredPos = new Vector3i(pos.getX(), 1, pos.getZ());
 
         // try and find somewhere in this region a spot to land
-        Region3i spawnArea = Region3i.createFromCenterExtents(desiredPos, ext);
-        Region worldRegion = getWorld().getWorldData(spawnArea);
+        Region3i searchArea = Region3i.createFromCenterExtents(desiredPos, ext);
+        org.terasology.world.generation.Region worldRegion = getWorld().getWorldData(searchArea);
         world = getWorld();
-
+        // graphs contains all graphs relevant within a radius of 7000 blocks from (0,0,0)
         GraphFacet graphs = worldRegion.getFacet(GraphFacet.class);
         WhittakerBiomeModelFacet model = worldRegion.getFacet(WhittakerBiomeModelFacet.class);
         Vector2f pos2d = new Vector2f(pos.getX(), pos.getZ());
-        CirclePickerClosest<org.terasology.polyworld.graph.Region> picker = new CirclePickerClosest<>(pos2d);
-
+        CirclePickerClosest<Region> picker = new CirclePickerClosest<>(pos2d);
+        boolean locationFound = false;
+        // searches for a spawn point such that it contains all the biomes required nearby
         for (Graph g : graphs.getAllGraphs()) {
             BiomeModel biomeModel = model.get(g);
-            for (org.terasology.polyworld.graph.Region r : g.getRegions()) {
+            for (Region r : g.getRegions()) {
                 WhittakerBiome biome = biomeModel.getBiome(r);
-                boolean flag = false;
-                for (org.terasology.polyworld.graph.Region neighbour : r.getNeighbors()) {
+                boolean ocean = false;
+                boolean forest = false;
+                boolean desert = false;
+                if (biomeModel.getBiome(r).equals(WhittakerBiome.TROPICAL_SEASONAL_FOREST) || biomeModel.getBiome(r).equals(WhittakerBiome.TEMPERATE_RAIN_FOREST) || biomeModel.getBiome(r).equals(WhittakerBiome.TEMPERATE_DECIDUOUS_FOREST) || biomeModel.getBiome(r).equals(WhittakerBiome.TROPICAL_RAIN_FOREST)) {
+                    forest = true;
+                }
+                for (Region neighbour : r.getNeighbors()) {
```

Comment:
why do the various regions need to be neighbors?
isn't it sufficient to go through all regions and check that all required biomes are present?

Reply:
This is written to ensure that all suitable biomes are present within a radius of about two regions from the starting region. Also if you have performance concerns regarding the nested loops, this can be better written as a bfs of depth two but it wouldn't have any visible difference due to the number of regions getting checked right now.

Reply:
I would argue that if we limit the island size during world generation we don't need to restrict the biomes being so close :thinking: But that's only my point of view.

Reply:
That would be better I agree. But as of now I don't know how that can be done. Also about two regions of radius doesn't compusorily include the beach/lakeshore biome (where the well is spawned).

Reply:
Don't know how _what_ can be done?
And I'm not sure I fully understand the comment about the beach 🤔